base/broadcast.jl

```diff
@@ -394,16 +390,6 @@ end
 ## Broadcasting utilities ##
 
 ## logic for deciding the BroadcastStyle
-# Dimensionality: computing max(M,N) in the type domain so we preserve inferrability
```

Comment:
Love that we don't need this anymore.

Reply:
It was an impressive bit of type gymnastics. Although since `ntuple` below was already `@generated`, marking `_max` as `@pure` might have been simpler.